aspnetcore/fundamentals/environments/3.1sample/EnvironmentsSample/StartupMethodConventions.cs

```diff
@@ -0,0 +1,112 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using System;
+
+namespace MethodConventions
+{
+    #region snippet
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        private void StartupConfigureServices(IServiceCollection services)
+        {
+            services.AddRazorPages();
+        }
+
+        public void ConfigureDevelopmentServices(IServiceCollection services)
+        {
+            MyTrace.TraceMessage();
+            StartupConfigureServices(services);
+        }
+
+        public void ConfigureStagingServices(IServiceCollection services)
+        {
+            MyTrace.TraceMessage();
+            StartupConfigureServices(services);
+        }
+
+        public void ConfigureProductionServices(IServiceCollection services)
+        {
+            MyTrace.TraceMessage();
+            StartupConfigureServices(services);
+        }
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            MyTrace.TraceMessage();
+            StartupConfigureServices(services);
+        }
+
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            MyTrace.TraceMessage();
+
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+            else
+            {
+                app.UseExceptionHandler("/Error");
+                app.UseHsts();
+            }
+
+            app.UseHttpsRedirection();
+            app.UseStaticFiles();
+
+            app.UseRouting();
+
+            app.UseAuthorization();
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapRazorPages();
+            });
+        }
+
+        public void ConfigureStaging(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            MyTrace.TraceMessage();
+
+            if (!env.IsStaging())
+            {
+                throw new Exception("Not staging.");
+            }
+
+            app.UseExceptionHandler("/Error");
+            app.UseHsts();
+
+            app.UseHttpsRedirection();
+            app.UseStaticFiles();
+
+            app.UseRouting();
+
+            app.UseAuthorization();
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapRazorPages();
+            });
+        }
+    }
+
+    public static class MyTrace
+    {
+        public static void 
+                     TraceMessage([System.Runtime.CompilerServices.CallerMemberName] 
```

Comment:
This is quite noisy without a fully qualified `CallerMemberName`. Can this just have `using` instead?

Reply:
I couldn't do that. See the [offical doc](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callermembernameattribute?view=netframework-4.8#examples) - they couldn't either.

Reply:
Oh yeah, that seems odd to me. It looks like pretty much all the types are fully qualified in those docs. Do you know the reason for it?

Reply:
I just tried it with your branch and it worked for me. Compiled and displayed the correct output. It's not a big deal, but it would've driven me mad if I hadn't tried to run it. I'm signing off now, but I'll be back around tomorrow if you need more from me here.

Reply:
The only way I could get it to work was to add `using CompilerServices = System.Runtime.CompilerServices;`